muspinsim/input/structure.py

```diff
@@ -159,118 +168,183 @@ def expand(self, offsets: List[ArrayLike]):
                             position=atom.position + offset,
                         )
                     )
-        self._atoms.extend(new_atoms)
-
-    def layer_expand(self, layers: int):
-        """Expands the structure outwards by duplicating it.
+        return new_atoms
 
-        Expands the structure outwards by duplicating it a specified number of
-        layers.
+    def _compute_layer_offsets(self, layer: int) -> List[ArrayLike]:
+        """Computes offsets for expanding this structure outwards
 
         Arguments:
-            layers {int} -- Number of layers to append e.g. if layers = 1,
-                            will expand along the x, y and z axes creating a
-                            structure 9 times larger than the original.
+            layer {int} -- Index of layer e.g. if layer = 1, will produce 8
+                           offsets along the x, y and z axes. layer = 2 will
+                           then produce another 26 offsets for expanding
+                           another layer outwards.
+        Returns:
+            offsets {List[ArrayLike]} -- List of computed offsets
         """
 
-        # Construct the offsets
-        x_values = np.arange(
-            -layers * self._unit_lengths[0],
-            (layers + 1) * self._unit_lengths[0],
-            self._unit_lengths[0],
-        )
-        y_values = np.arange(
-            -layers * self._unit_lengths[1],
-            (layers + 1) * self._unit_lengths[1],
-            self._unit_lengths[1],
-        )
-        z_values = np.arange(
-            -layers * self._unit_lengths[2],
-            (layers + 1) * self._unit_lengths[2],
-            self._unit_lengths[2],
-        )
-        offsets = np.array(np.meshgrid(x_values, y_values, z_values)).T.reshape(-1, 3)
-        # Remove 0, 0, 0
-        offsets = np.delete(offsets, int(np.floor(len(offsets) / 2)), axis=0)
+        # Compute all potential values for the x, y and z axes
+        potential_values = [
+            np.arange(
+                -layer * self._unit_lengths[i],
+                (layer + 1) * self._unit_lengths[i],
+                self._unit_lengths[i],
+            )
+            for i in range(3)
+        ]
 
-        # Now expand for each
-        self.expand(offsets)
+        # Obtain the extremes and values in between for each axis
+        extreme_values = [
+            [potential_values[i][0], potential_values[i][-1]] for i in range(3)
+        ]
+        middle_values = [potential_values[i][1:-1:] for i in range(3)]
 
-    def compute_distances(self):
-        """Computes the vectors and distances between the muon and every atom.
+        # Compute combinations
+        offsets = []
 
-        Computes the vectors and distances between the muon and every atom.
-        """
-        muon = self._atoms[self._muon_index]
+        # Hold z fixed at one of the extreme values and vary others
```

Comment:
I was originally going to try remove them from the arrays themselves but thought this would be better for the same reasons. Never used `set` before so its useful to know.